### PR TITLE
Change License to GNU

### PR DESCRIPTION
**Pull request by jparisu**
License should meet Qt GNU GPL license

Signed-off-by: jparisu <javierparis@eprosima.com>

### CODE REVIEW

**Comment by jparisu** [comment]
TODO:
- [ ] Change documents internal license 
- [ ] Change license references (documentation, etc.)

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,16 +1,19 @@
-# Copyright 2017 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+# Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
 #
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
+# This file is part of eProsima Fast DDS Shapes-Demo.
```
Comment (JLBuenoLopez-eProsima) [inline CMakeLists.txt:3]:
```suggestion
# This file is part of eProsima Fast DDS Shapes Demo.
```

Seeing the README and the documentation, it seems to be written without the hyphen. Please, apply to all this PR.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,16 +1,19 @@
-# Copyright 2017 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+# Copyright 2021 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (JLBuenoLopez-eProsima) [inline CMakeLists.txt:1]: Is it really necessary to change the copyright?

File: include/eprosimashapesdemo/qt/DrawArea.h [hunk 3]
```diff
@@ -1,16 +1,36 @@
...
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// eProsima Fast DDS Shapes-Demo is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with eProsima Fast DDS Shapes-Demo. If not, see <https://www.gnu.org/licenses/>.
```
Comment (JLBuenoLopez-eProsima) [inline include/eprosimashapesdemo/qt/DrawArea.h:33]: It is repeated. Please remove.

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
Please rebase

File: CMakeLists.txt (hunk 1 above)
Comment (jparisu) [reply CMakeLists.txt:3]: attempt in commit 144eebb

File: CMakeLists.txt (hunk 2 above)
Comment (jparisu) [reply CMakeLists.txt:1]: I would take advantage of this PR to update them all (I think that they should be updated as the files are changed).

File: src/shapesdemo/Shape.cxx [hunk 4]
```diff
@@ -127,11 +126,6 @@ size_t ShapeType::getCdrSerializedSize(
     size_t initial_alignment = current_alignment;
 
 
-    current_alignment += 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4) + data.color().size() + 1;
-
-    current_alignment += 4 + eprosima::fastcdr::Cdr::alignment(current_alignment, 4);
-
-
```
Comment (JLBuenoLopez-eProsima) [inline src/shapesdemo/Shape.cxx:134]: I think these lines have been deleted by mistake.

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM! :sweat_smile: I did not realize before.

**Pull request merged**